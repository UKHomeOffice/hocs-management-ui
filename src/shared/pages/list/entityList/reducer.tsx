import { State } from './state';
import { Action } from './actions';

export const reducer = (state: State, action: Action): State => {
    switch (action.type) {
        case 'PopulateEntities':
            return { ...state, entities: action.payload, entitiesLoaded: true };
    }
    return state;
};
