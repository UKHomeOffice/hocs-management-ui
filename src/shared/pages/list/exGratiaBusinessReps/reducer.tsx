import { State } from './state';
import { Action } from './actions';

export const reducer = (state: State, action: Action): State => {
    switch (action.type) {
        case 'PopulateRepresentatives':
            return { ...state, representatives: action.payload, representativesLoaded: true };
    }
    return state;
};
