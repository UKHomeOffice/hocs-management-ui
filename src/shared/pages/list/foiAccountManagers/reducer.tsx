import { State } from './state';
import { Action } from './actions';

export const reducer = (state: State, action: Action): State => {
    switch (action.type) {
        case 'PopulateAccountManagers':
            return { ...state, accountManagers: action.payload, accountManagersLoaded: true };
    }
    return state;
};
