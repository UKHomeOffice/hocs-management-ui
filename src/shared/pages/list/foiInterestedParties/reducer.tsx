import { State } from './state';
import { Action } from './actions';

export const reducer = (state: State, action: Action): State => {
    switch (action.type) {
        case 'PopulateParties':
            return { ...state, parties: action.payload, partiesLoaded: true };
    }
    return state;
};
