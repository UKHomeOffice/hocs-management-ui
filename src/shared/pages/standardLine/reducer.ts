import InputEventData from '../../models/inputEventData';
import StandardLine from '../../models/standardLine';

export const reducer = (state: StandardLine, inputEventData: InputEventData) => {
    const newState = { ...state, [inputEventData.name]: inputEventData.value };
    return newState;
};
