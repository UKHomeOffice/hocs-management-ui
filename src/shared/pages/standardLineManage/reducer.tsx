import { State } from './state';
import { Action } from './actions';
import StandardLineModel from 'shared/models/standardLineModel';

export const reducer = (state: State, action: Action): State => {
    switch (action.type) {
        case 'PopulateStandardLines':
            return { ...state, allStandardLines: action.payload, activeStandardLines: filterLines(action.payload, state.filter.toUpperCase(), state.excludeExpired), standardLinesLoaded: true };
        case 'FilterStandardLines':
            return { ...state, filter: action.payload, activeStandardLines: filterLines(state.allStandardLines, action.payload.toUpperCase(), state.excludeExpired) };
        case 'ExcludeExpiredCheckTrigger':
            return { ...state, excludeExpired: action.payload, activeStandardLines: filterLines(state.allStandardLines, state.filter.toUpperCase(), action.payload) };
    }
    return state;
};

const filterLines = (standardLines: StandardLineModel[], filter: string, excludeExpired: boolean): StandardLineModel[] => {
    let filteredStandardLines = standardLines;
    if (standardLines && filter) {
        filteredStandardLines = [];
        for (let i = 0; i < standardLines.length; i += 1) {
            const standardLine = standardLines[i];
            if (doesFilterMatchValue(standardLine.topic, filter) || doesFilterMatchValue(standardLine.displayName, filter) || doesFilterMatchValue(standardLine.expiryDate, filter)) {
                filteredStandardLines.push(standardLine);
            }
        }
    }
    return excludeExpired ? removeExpiredStarndardLines(filteredStandardLines) : filteredStandardLines;
};

const doesFilterMatchValue = (value: string, filter: string) => {
    return value && filter && value.toUpperCase && value.toUpperCase().indexOf(filter) !== -1;
};

const removeExpiredStarndardLines = (standardLines: StandardLineModel[]): StandardLineModel[] => {
    if (standardLines) {
        const filteredStandardLines = [];
        for (let i = 0; i < standardLines.length; i += 1) {
            const standardLine = standardLines[i];
            if (!standardLine.isExpired) {
                filteredStandardLines.push(standardLine);
            }
        }
        return filteredStandardLines;
    }
    return standardLines;
};
