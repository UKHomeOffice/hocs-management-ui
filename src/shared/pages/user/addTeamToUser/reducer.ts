import Item from '../../../models/item';
import { User } from '../../../models/user';

// state
export const initialState: State = {
    user: undefined,
    selectedTeam: undefined,
    selectedTeams: []
};
export interface State {
    user?: User;
    selectedTeam?: Item | undefined;
    selectedTeams: Item[];
}

// actions
type AddToSelectedTeams = {
    payload: Item;
    type: 'AddToSelectedTeams';
};
type SetUser = {
    payload?: User;
    type: 'SetUser';
};
type ClearSelectedTeam = {
    type: 'ClearSelectedTeam';
};
type RemoveFromSelectedTeams = {
    payload: Item;
    type: 'RemoveFromSelectedTeams';
};

export type Action =
    AddToSelectedTeams |
    SetUser |
    ClearSelectedTeam |
    RemoveFromSelectedTeams;

// reducer
export const reducer = (state: State, action: Action) => {
    switch (action.type){
        case 'AddToSelectedTeams':
            return { ...state, selectedTeams: [...state.selectedTeams, action.payload] };
        case 'SetUser':
            return { ...state, user: action.payload  };
        case 'ClearSelectedTeam':
            return { ...state, selectedTeam: undefined };
        case 'RemoveFromSelectedTeams':
            return { ...state, selectedTeams: [...state.selectedTeams.filter(team => team.value !== action.payload.value)] };
    }
    return state;
};
