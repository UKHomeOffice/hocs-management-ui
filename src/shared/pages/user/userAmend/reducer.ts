import InputEventData from '../../../models/inputEventData';
import State from './state';

export const reducer = (state: State, inputEventData: InputEventData) => {
    const newState = { ...state, [inputEventData.name]: inputEventData.value };
    return newState;
};
