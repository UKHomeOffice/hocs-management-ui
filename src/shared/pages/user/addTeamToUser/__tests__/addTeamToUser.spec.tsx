import React from 'react';
import * as useError from '../../../../hooks/useError';
import { State } from '../reducer';
import { match, MemoryRouter } from 'react-router-dom';
import { createBrowserHistory, History, Location } from 'history';
import * as TeamsService from '../../../../services/teamsService';
import * as UsersService from '../../../../services/usersService';
import AddTeamToUser from '../addTeamToUser';
import { act, fireEvent, getByText, render, RenderResult, wait } from '@testing-library/react';
import ErrorMessage from '../../../../models/errorMessage';
import { EMPTY_TEAMS_SUBMIT_ERROR_DESCRIPTION, EMPTY_TEAMS_SUBMIT_ERROR_TITLE, GENERAL_ERROR_TITLE, LOAD_TEAMS_ERROR_DESCRIPTION, LOAD_USER_ERROR_DESCRIPTION } from '../../../../models/constants';

let match: match<any>;
let history: History<any>;
let location: Location;
let mockState: State;

jest.mock('../../../../services/teamsService', () => ({
    __esModule: true,
    getTeams: jest.fn().mockReturnValue(Promise.resolve([
        {
            label: '__team1__',
            value: '__teamId1__'
        },
        {
            label: '__team2__',
            value: '__teamId2__'
        },
        {
            label: '__team3__',
            value: '__teamId3__'
        },
        {
            label: '__team4__',
            value: '__teamId4__'
        }
    ]))
}));
jest.mock('../../../../services/usersService', () => ({
    __esModule: true,
    addUsersToTeam: jest.fn().mockReturnValue(Promise.resolve()),
    getUser: jest.fn().mockReturnValue(Promise.resolve({
        label: '__user__',
        value: '__userId__'
    }))
}));

const getTeamsSpy = jest.spyOn(TeamsService, 'getTeams');
const addUserToTeamSpy = jest.spyOn(UsersService, 'addUsersToTeam');
const getUserSpy = jest.spyOn(UsersService, 'getUser');
const useReducerSpy = jest.spyOn(React, 'useReducer');
const reducerDispatch = jest.fn();
const useErrorSpy = jest.spyOn(useError, 'default');
const addFormErrorSpy = jest.fn();
const clearErrorsSpy = jest.fn();
const setMessageSpy = jest.fn();

const renderComponent = () => render(
    <MemoryRouter>
        <AddTeamToUser history={history} location={location} match={match}/>
    </MemoryRouter>
);

beforeEach(() => {
    history = createBrowserHistory();
    match = {
        isExact: true,
        params: { userId: '__userId__' },
        path: '',
        url: ''
    };

    location = {
        hash: '',
        key: '',
        pathname: '',
        search: '',
        state: {}
    };
    mockState = {
        user: {
            id: '__user__',
            username: 'd',
            email: 'd',
            firstName: 'd',
            lastName: 'd',
            enabled: false
        },
        selectedTeam: {
            label: '__team1__',
            value: '__teamId__'
        },
        selectedTeams: [{
            label: '__team1__',
            value: '__teamId1__'
        }, {
            label: '__team2__',
            value: '__teamId2__'
        }]
    };
    useReducerSpy.mockImplementationOnce(() => [mockState, reducerDispatch]);
    useErrorSpy.mockImplementation(() => [{}, addFormErrorSpy, clearErrorsSpy, setMessageSpy]);
    reducerDispatch.mockReset();
    addFormErrorSpy.mockReset();
    clearErrorsSpy.mockReset();
    setMessageSpy.mockReset();
});

describe('when the addTeamToUser component is mounted', () => {
    it('should render', async () => {
        let wrapper: RenderResult;
        act(() => {
            wrapper = renderComponent();
        });

        await wait(() => {
            expect(getUserSpy).toHaveBeenCalled();
            expect(getTeamsSpy).toHaveBeenCalled();
            expect(wrapper.container).toMatchSnapshot();
        });
    });

    it('should initially render null before the user is returned', async () => {
        let wrapper: RenderResult;
        getUserSpy.mockReturnValueOnce(Promise.resolve({
            id: '__user__',
            username: '__userId__',
            firstName: 'd',
            lastName: 'd',
            email: 'd',
            password: 'ewrwe',
            enabled: false
        }));

        act(() => {
            wrapper = renderComponent();
        });

        await wait(() => {
            expect(wrapper.container.outerHTML).toMatchSnapshot();
        });
    });

    it('should display an error if the call to retrieve the user fails', async () => {
        expect.assertions(1);
        getUserSpy.mockImplementation(() => Promise.reject('error'));

        act(() => {
            renderComponent();
        });

        await wait(() => {
            expect(setMessageSpy).toBeCalledWith(new ErrorMessage(LOAD_USER_ERROR_DESCRIPTION, GENERAL_ERROR_TITLE));
        });
    });

    it('should display an error if the call to retrieve the teams fails', async () => {
        expect.assertions(1);
        getTeamsSpy.mockImplementation(() => Promise.reject('error'));

        act(() => {
            renderComponent();
        });

        await wait(() => {
            expect(setMessageSpy).toBeCalledWith(new ErrorMessage(LOAD_TEAMS_ERROR_DESCRIPTION, GENERAL_ERROR_TITLE));
        });

    });

});

describe('when the submit button is clicked', () => {
    let wrapper: RenderResult;
    beforeEach(() => {
        act(() => {
            wrapper = renderComponent();
        });
    });

    it('should call the service and dispatch actions for the selected options', async () => {
        expect.assertions(3);

        act(() => {
            const submitButton = getByText(wrapper.container, 'Add selected teams');
            fireEvent.click(submitButton);
        });

        await wait(async () => {
            expect(addUserToTeamSpy).nthCalledWith(1, [{
                label: 'd',
                value: '__user__'
            }], '__teamId1__');
            expect(addUserToTeamSpy).nthCalledWith(2, [{
                label: 'd',
                value: '__user__'
            }], '__teamId2__');
            expect(clearErrorsSpy).toBeCalled();
        });
    });

    it('should display errors for each erroring team member add request', async () => {
        addUserToTeamSpy.mockImplementationOnce(() => Promise.reject('Error'));

        act(() => {
            const submitButton = getByText(wrapper.container, 'Add selected teams');
            fireEvent.click(submitButton);
        });

        await wait(() => {
            expect(setMessageSpy).toHaveBeenCalled();
        });
    });

    it('should set an error when no users are selected', async () => {
        mockState.selectedTeams = [];
        act(() => {
            const submitButton = getByText(wrapper.container, 'Add selected teams');
            fireEvent.click(submitButton);
        });

        await wait(async () => {
            expect(setMessageSpy).toBeCalledWith(new ErrorMessage(EMPTY_TEAMS_SUBMIT_ERROR_DESCRIPTION, EMPTY_TEAMS_SUBMIT_ERROR_TITLE));
        });
    });
});

describe('when the remove button is clicked', () => {
    it('should remove the row from the selected users collection', async () => {

        let wrapper: RenderResult;
        act(() => {
            wrapper = renderComponent();
        });

        await wait(async () => {
            const selectedTeam = getByText(wrapper.container, '__team2__');
            const row = (selectedTeam.closest('tr'));
            const removeButton = getByText(row as HTMLElement, 'Remove');
            fireEvent.click(removeButton);
        });

        expect(reducerDispatch).toBeCalledWith({ type: 'RemoveFromSelectedTeams', payload: { label: '__team2__', value: '__teamId2__' } });
    }, 20000);
});
