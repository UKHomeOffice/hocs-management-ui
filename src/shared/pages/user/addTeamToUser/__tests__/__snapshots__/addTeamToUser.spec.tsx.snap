// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the addTeamToUser component is mounted should initially render null before the user is returned 1`] = `"<div><div class=\\"govuk-form-group\\"><a class=\\"govuk-back-link\\" href=\\"/user-view/__userId__\\">Back</a><h1 class=\\"govuk-heading-xl\\">Add Teams</h1><h2 class=\\"govuk-heading-l\\">User: d d (d)</h2><div class=\\"govuk-form-group\\"><label for=\\"select-teams-input\\" id=\\"select-teams-label\\" class=\\"govuk-label govuk-label--s\\">Select teams</label><div class=\\" css-2b097c-container\\" id=\\"select-teams\\"><div class=\\"govuk-typeahead__control css-1g8ily9-Control\\"><div class=\\"govuk-typeahead__value-container govuk-typeahead__value-container--has-value css-pilx6v-ValueContainer\\"><div class=\\"govuk-typeahead__single-value css-1uccc91-singleValue\\">__team1__</div><div class=\\"css-b8ldur-Input\\"><div class=\\"govuk-typeahead__input\\" style=\\"display: inline-block;\\"><input autocapitalize=\\"none\\" autocomplete=\\"off\\" autocorrect=\\"off\\" id=\\"select-teams-input\\" spellcheck=\\"false\\" tabindex=\\"0\\" type=\\"text\\" aria-autocomplete=\\"list\\" style=\\"box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;\\" value=\\"\\"><div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;\\"></div></div></div></div><div class=\\"govuk-typeahead__indicators css-1hb7zxy-IndicatorsContainer\\"><div aria-hidden=\\"true\\" class=\\"govuk-typeahead__indicator govuk-typeahead__clear-indicator css-tlfecz-indicatorContainer\\"><svg height=\\"20\\" width=\\"20\\" viewBox=\\"0 0 20 20\\" aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"css-6q0nyr-Svg\\"><path d=\\"M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z\\"></path></svg></div><span class=\\"govuk-typeahead__indicator-separator css-62wm9a-IndicatorSeparator\\"></span><div aria-hidden=\\"true\\" class=\\"govuk-typeahead__indicator govuk-typeahead__dropdown-indicator css-hbt8bd-DropdownIndicator\\"><svg height=\\"20\\" width=\\"20\\" viewBox=\\"0 0 20 20\\" aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"css-6q0nyr-Svg\\"><path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\"></path></svg></div></div></div></div></div><div class=\\"govuk-grid-row\\"><div class=\\"govuk-grid-column-two-thirds-from-desktop\\"><table class=\\"govuk-table\\"><caption class=\\"govuk-table__caption\\">Teams to be added</caption><tbody class=\\"govuk-table__body\\"><tr class=\\"govuk-table__row\\"><th scope=\\"row\\" class=\\"govuk-table__header\\">__team1__</th><td class=\\"govuk-table__cell\\"><a class=\\"govuk-link\\" href=\\"#\\">Remove<span class=\\"govuk-visually-hidden\\"> user</span></a></td></tr><tr class=\\"govuk-table__row\\"><th scope=\\"row\\" class=\\"govuk-table__header\\">__team2__</th><td class=\\"govuk-table__cell\\"><a class=\\"govuk-link\\" href=\\"#\\">Remove<span class=\\"govuk-visually-hidden\\"> user</span></a></td></tr></tbody></table></div></div><button type=\\"submit\\" class=\\"govuk-button view-team-button\\">Add selected teams</button></div></div>"`;

exports[`when the addTeamToUser component is mounted should render 1`] = `
<div>
  <div
    class="govuk-form-group"
  >
    <a
      class="govuk-back-link"
      href="/user-view/__userId__"
    >
      Back
    </a>
    <h1
      class="govuk-heading-xl"
    >
      Add Teams
    </h1>
    <h2
      class="govuk-heading-l"
    >
      User: d d (d)
    </h2>
    <div
      class="govuk-form-group"
    >
      <label
        class="govuk-label govuk-label--s"
        for="select-teams-input"
        id="select-teams-label"
      >
        Select teams
      </label>
      <div
        class=" css-2b097c-container"
        id="select-teams"
      >
        <div
          class="govuk-typeahead__control css-1g8ily9-Control"
        >
          <div
            class="govuk-typeahead__value-container govuk-typeahead__value-container--has-value css-pilx6v-ValueContainer"
          >
            <div
              class="govuk-typeahead__single-value css-1uccc91-singleValue"
            >
              __team1__
            </div>
            <div
              class="css-b8ldur-Input"
            >
              <div
                class="govuk-typeahead__input"
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="select-teams-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class="govuk-typeahead__indicators css-1hb7zxy-IndicatorsContainer"
          >
            <div
              aria-hidden="true"
              class="govuk-typeahead__indicator govuk-typeahead__clear-indicator css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-6q0nyr-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                />
              </svg>
            </div>
            <span
              class="govuk-typeahead__indicator-separator css-62wm9a-IndicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="govuk-typeahead__indicator govuk-typeahead__dropdown-indicator css-hbt8bd-DropdownIndicator"
            >
              <svg
                aria-hidden="true"
                class="css-6q0nyr-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="govuk-grid-row"
    >
      <div
        class="govuk-grid-column-two-thirds-from-desktop"
      >
        <table
          class="govuk-table"
        >
          <caption
            class="govuk-table__caption"
          >
            Teams to be added
          </caption>
          <tbody
            class="govuk-table__body"
          >
            <tr
              class="govuk-table__row"
            >
              <th
                class="govuk-table__header"
                scope="row"
              >
                __team1__
              </th>
              <td
                class="govuk-table__cell"
              >
                <a
                  class="govuk-link"
                  href="#"
                >
                  Remove
                  <span
                    class="govuk-visually-hidden"
                  >
                     user
                  </span>
                </a>
              </td>
            </tr>
            <tr
              class="govuk-table__row"
            >
              <th
                class="govuk-table__header"
                scope="row"
              >
                __team2__
              </th>
              <td
                class="govuk-table__cell"
              >
                <a
                  class="govuk-link"
                  href="#"
                >
                  Remove
                  <span
                    class="govuk-visually-hidden"
                  >
                     user
                  </span>
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <button
      class="govuk-button view-team-button"
      type="submit"
    >
      Add selected teams
    </button>
  </div>
</div>
`;
